vuepress/docs/docs/client.md

```diff
@@ -32,16 +32,70 @@ addSbtPlugin("com.github.ghostdogpr" % "caliban-codegen-sbt" % "1.0.1")
 
 And enable it in your `build.sbt` file:
 ```scala
-enablePlugins(CodegenPlugin)
+enablePlugins(CalibanPlugin)
 ```
 
-Then call the `calibanGenClient` sbt command.
+### File generation
+
+At this point, the `caliban` command will cause any files in `src/main/graphql` to be translated into a Caliban-generated client library. This happens automatically any time you `compile`.
+
+By default, all clients are generated with the same client name as the source file, in the `caliban` top-level package.
+
+In order to supply more configuration options to the code generator, you can use the `calibanSettings` sbt setting, combined with the `calibanSetting` function to scope the settings to a particular file:
+```scala
+      // The `file("Service.graphql")` is a path suffix for some file in `src/main/graphql`
+      Compile / caliban / calibanSettings += calibanSetting(file("Service.graphql"))(
+        cs =>
+          cs.packageName("com.example.graphql.client")
+            .scalarMapping(
+              "LanguageCode" -> "com.example.models.LanguageCode",
+            )
+            .scalarMapping(
+              "Timestamp" -> "java.sql.Timestamp",
+              "DayOfWeek" -> "java.time.DayOfWeek",
+              "IntRange"  -> "com.github.tminglei.slickpg.Range[Int]"
+            )
+            .imports("com.example.graphql.client.implicits._")
+      )
+```
+
+### URL generation
+
+The `calibanSetting` function also permits generating clients for supplied `url`'s:
+```scala
+      Compile / caliban / calibanSettings += calibanSetting(url("http://my-example-service/graphql"))(
+        cs =>
+          cs.clientName("ExampleServiceClient")
+            .packageName("com.example.graphql.client")
+      )
+```
+
+### `calibanSetting` config parameters
+
+The settings available on the `cs` (`CalibanSettings`) builder are:
+```
+  def scalafmtPath(path: String): CalibanSettings
+  def packageName(name: String): CalibanSettings
+  def genView(value: Boolean): CalibanSettings
+  def effect(tpe: String): CalibanSettings
+  def scalarMapping(mapping: (String,String)*): CalibanSettings
+  def imports(values: String*): CalibanSettings
+
+  // Only defined for `url` settings, for use in supplying extra headers when fetching the schema itself
+  def headers(pairs: (String,String)*): CalibanSettings
+```
+
+### `calibanGenClient`
+
+Should you prefer or otherwise need the previous style of manual client codegen, supplied by the `calibanGenClient` function, that command is still available. Documentation for that command is as follows:
```

Comment:
Maybe no need to refer to the "previous style" which is unclear for newcomers, but instead just say something like "If you prefer to generate the client explicitly rather than automatically, ..."